### ISSUE TITLE
Weak ETags on non-GET requests (Trac #116)

### ISSUE THREAD

**Issue by mnot** [httpwg/httpbis-issues#116]
RFC2616 limits the use of weak ETags to full (non-subrange) GET requests. 

While the limitation on ranged requests makes sense, weak ETag validation is useful on full non-GET requests; e.g., PUT with If-Match.

Migrated from https://trac.ietf.org/ticket/116
```json
{
    "status": "closed",
    "changetime": "2009-07-29T11:24:04",
    "_ts": "1248866644000000",
    "description": "RFC2616 limits the use of weak ETags to full (non-subrange) GET requests. \n\nWhile the limitation on ranged requests makes sense, weak ETag validation is useful on full non-GET requests; e.g., PUT with If-Match.",
    "reporter": "mnot@pobox.com",
    "cc": "",
    "resolution": "fixed",
    "time": "2008-05-09T05:39:05",
    "component": "p4-conditional",
    "summary": "Weak ETags on non-GET requests",
    "priority": "urgent",
    "keywords": "",
    "milestone": "08",
    "owner": "julian.reschke@gmx.de",
    "type": "design",
    "severity": "Active WG Document"
}
```

**Comment by mnot** [comment]
Trac update at 20080509T14:36:38: 
* julian.reschke@gmx.de changed attachment from "" to "i116.diff"
* julian.reschke@gmx.de commented:

> Proposed patch to part 4.

**Comment by mnot** [comment]
Trac update at 20080515T11:38:24: 
* julian.reschke@gmx.de commented:

> Fixed in [245]:
> 
> Resolve #116: allow weak entity tags in all requests except range requests (closes #116).


* julian.reschke@gmx.de changed resolution from "" to "fixed"
* julian.reschke@gmx.de changed status from "new" to "closed"

**Comment by mnot** [comment]
Trac update at 20090729T10:51:36: 
* julian.reschke@gmx.de commented:

> Section about If-Match needs to be updated as well.


* julian.reschke@gmx.de changed priority from "" to "normal"
* julian.reschke@gmx.de changed resolution from "fixed" to ""
* julian.reschke@gmx.de changed status from "closed" to "reopened"

**Comment by mnot** [comment]
Trac update at 20090729T10:51:53: 
* julian.reschke@gmx.de changed milestone from "unassigned" to "08"
* julian.reschke@gmx.de changed owner from "" to "julian.reschke@gmx.de"
* julian.reschke@gmx.de changed priority from "normal" to "urgent"
* julian.reschke@gmx.de changed status from "reopened" to "new"

**Comment by mnot** [comment]
Trac update at 20090729T11:24:04: 
* julian.reschke@gmx.de commented:

> Fixed in [656]:
> 
> Resolve #116: If-Match was still requiring strong matches (closes #116)


* julian.reschke@gmx.de changed resolution from "" to "fixed"
* julian.reschke@gmx.de changed status from "new" to "closed"
